libraries/AC_Avoidance/AP_OAPathPlanner.cpp

```diff
@@ -180,15 +212,31 @@ AP_OAPathPlanner::OA_RetState AP_OAPathPlanner::mission_avoidance(const Location
 void AP_OAPathPlanner::avoidance_thread()
 {
     while (true) {
+#if !HAL_MINIMIZE_FEATURES
+        if (_oadatabase.queue_count() > 0) {
+            // database needs attention, bump up the speed
+            _oadatabase.update();
+            hal.scheduler->delay(1);
+        } else {
+            // database needs little only a attention, slow things down
+            _oadatabase.update();
+            hal.scheduler->delay(20);
+        }
 
-        // run at 10hz or less
+        const uint32_t now = AP_HAL::millis();
```

Comment:
Again, not a blocker but let's remove the two lines below:

> hal.scheduler->delay(100);
> const uint32_t now = AP_HAL::millis();

by moving HAL_MINIMIZE_FEATURES's endif up and always continuing if less than 100ms has passed since the last time we ran bendyrule or dijkstra's.  I think the new method is better because it runs the code every 100ms instead of 100ms + processing time.

Reply:
Ah, I see the issue is that you want to avoid a busy loop.  What we could do is instead calculate a delay time which is 1ms, 20ms or the time remaining until we need to run bendyrule or dijkstra's (i.e. 100ms - (now - avoidance_latest_ms))

Reply:
good ideas for another PR